### DISCUSSION TITLE
Suggestions for our Discord?

### DISCUSSION THREAD

**Discussion by Turbotailz** [LuckPerms/LuckPerms#2826] [General]
The LP discord has been running pretty smoothly for a few years now. Which is great for a plugin support platform, but can it be better?

Recently we just reached 10k members, which is a great milestone, however the retention rate is actually quite low - due to users mainly joining for support then leaving once they find their answer (or not, sadly not everyone has a happy ending, can't please em all).

![image](https://user-images.githubusercontent.com/10846855/104387347-ffc6d280-558a-11eb-956d-e96df67777b4.png)

I'm opening a discussion here to talk about what we could possibly do to the LP discord to make it a better place for people to seek support while still being able to maintain some form of community with the large amount of members we have.

Do you have anything to suggest? Leave it here! We can't promise everything will be actioned but at least we can talk about it, without it getting buried in the channels 😉

**Comment by ItsCryptic** [comment]
It's another staff role "pretty unlucky (admin)", this role would be given to the mods who help out in the community a lot (recognizing them). I'm thinking atm just off the top of my head who I see around often Larry and Fefo (the others may be around but in different timezones that I'm not aware of so idk). I'm not sure exactly what mods do and don't have access too. So I couldn't even tell you it's responsibilities.

**Comment by tadhgboyle** [comment]
**Rename `general-1` and `general-2` to `support-1` and `support-2`, and create `luckperms`.**

About once a day I see people ask where to get support, which wastes their time and "ours".
Even if someone wasn't looking for support, it just makes more sense.

An example of how it works at NamelessMC:
`nameless` - general discussion about NamelessMC. Ideas, update hype, etc.
`support-1` - support for NamelessMC
`support-2` - support for NamelessMC

This could be used for the LuckPerms guild as well.
It would clear confusion of channel names, and wont get general discussion about LuckPerms get stuck in the busy support channels.

**Comment by lucko** [comment]
## News & Updates

We've always just posted news as regular messages to the #news channel, without ever tagging @ everyone. Prior to Discord's publish feature, these messages (obviously) weren't published either!

I wonder whether this current approach is limiting the reach of our (obviously always very cool & very important) news posts.

Some strategies taken by other Discord servers:

* Creating a role which users can subscribe to to be tagged when updates happen

I think this could work, maybe?! Would people voluntarily subscribe tho?

* Tagging @ everyone

Tbh, I'm terrified of the backlash we'd probably get if we started doing this out of the blue 😆 

Or maybe we should just keep things as-is.

I'm open to other ideas!

**Reply by Turbotailz** [reply]
I think doing @everyone in a server with over 10k members is a no-brainer - don't do it!

A pingable role is a nice alternative :)

**Reply by ItsCryptic** [reply]
yes please make a pingable role, pinging 10k members doesn't always work. Discord seems to have issues delivering pings after like ~500 members or so. So the people who do want updates may not receive them.

**Reply by Virologicbody** [reply]
I agree with the pingable role.
Personally I just look out for announcements regarding updates etc.

I've seen in other Discord servers members were complaining about being pinged, although they obviously haven't learned how to 'mute' a server, so I think having a pingable role would be a more cautious approach.

**Comment by Laarryy** [comment]
New channel scheme suggestion I made in the discord which everyone I have asked largely agrees with (including staff/helpers) - posted at the request of turbo (thx turbo UwU <3) 

#support
#support-2
#luckperms
#off-topic
#dev-api
#lp-web (can also be merged with #luckperms)
#translators

**Reply by ItsCryptic** [reply]
you forgot patreons but other than that yes I approve

**Reply by tadhgboyle** [reply]
Yes but also make #pebkac public

**Reply by Laarryy** [reply]
> you forgot patreons but other than that yes I approve

> Yes but also make #pebkac public

Private channels stay private and where they are - no point changing those as they affect the minority of users and do what they should anyways

**Comment by Turbotailz** [comment]
It would be nice to have the banner back. Incentives for boosting? Maybe access to our Patreon Minecraft server?

**Comment by JasonHorkles** [comment]
New clippy commands:
!ab - Tells people to remove EssxAntiBuild and any other modules they don't need
!1l - "Discord supports complete sentences. Less lines more content, please."

**Comment by BomBardyGamer** [comment]
Maybe we should expand LuckPerms to not just be a hub for LuckPerms, but also a hub for all of @lucko 's other projects as well, and also a hub for developers to talk and help each other out, a bit like how HelpChat works. I'm not saying turn this into every other developer Discord server (because no other Discord server has lucko 🙂), but maybe just add some stuff so people can get support with developement and other stuff that lucko's made.

lucko isn't just LuckPerms, he's made so many other tools to help out developers as well, and we should encourage the use of them

**Comment by PiggyPiglet** [comment]
I've been making a bot over the last month or so called [DocDex](https://github.com/PiggyPiglet/DocDex), which allows you to view javadoc objects in discord. It could be beneficial to the discord when providing support for the api, and I've already added the luckperms api to the public instance.

[Invite](https://piggypiglet.me/docdex)

pics:
![image](https://user-images.githubusercontent.com/11957313/106298341-bcf54200-628e-11eb-8278-a3528225bde1.png)
![image](https://user-images.githubusercontent.com/11957313/106298722-342ad600-628f-11eb-82b2-d6092a77cd1b.png)

**Reply by lucko** [reply]
This is really cool

How do we get it added?

**Reply by Laarryy** [reply]
Just click the invite link in Piggy's post, and ask it nicely to join LP!

**Reply by PiggyPiglet** [reply]
Yeah like larry said, there’s an invite link in the message

**Comment by Parthurax** [comment]
I searched for a place where you can ask everything about server related Minecraft. The are just a few places where are some experts, who really understand what they are doing.

So it would be cool to find a place where you can ask everything and know that there a people who dont talk about things they dont understand.

And i dont have things in mind like, i cant read a wiki and dont understand english at all, question.

Sometimes you have dumb questions and thats okay but where you go to ask about a secure linux setting related to minecraft, how you secure bungeecord or whatever. Most of the documentation is very old, and one big community is split into thousands of discords, most of them never help each other.

Since luckperms is a thing that just works, and is well documented, it would be cool if you turn into a place for all minecraft server admins, and not just a place for luckperms.

**Reply by Turbotailz** [reply]
We made a Discord server for exactly this sort of thing, it's called Syscraft, you are welcome to join - https://discord.gg/Dx6SSkx

I think it's best we keep the LP discord to mainly LP stuff, although we do have a #general channel where you can ask these sorts of questions, it's probably better to ask a different community as you'll find more quality in answers there I believe.

**Comment by Chew** [comment]
Rory feed

**Comment by ghost** [comment]
Add a role called 'Volunteer Helper' that perhaps people can give themselves via reaction with no perks other than a different color and perhaps showing up right under the official helpful role.

Pros: 
- Users who know more than the average or simply have the hang of setting up lp can volunteer to show themselves as potentially helpful person, obviously not everyone knows everything, but it's worth a try.
- Helpful people would perhaps get less pings if there was a volunteer role. Basically expose ourselves to be pinged xD
- As end-user, seeing as there are not too many 'helpful' roled people may make users reluctant to asking questions, thinking that the support team is rather small compared to the actual size of the discord.

Cons:
- People who may think they know a lot about luckperms, giving themselves the role, could potentially give support-seeking users wrong advice but honestly this already happens from time to time between regular users without suggested role.

I see some people who generally like to answer questions who keep returning to do so. Giving them a way to make it more visible that they like to help may help the community to represent stability and a larger support team.
